src/Surfnet/StepupSelfService/SamlStepupProviderBundle/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,153 @@
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('surfnet_stepup_self_service_saml_stepup_provider');
+
+        $this->addRoutesSection($rootNode);
+        $this->addProvidersSection($rootNode);
+
+        return $treeBuilder;
+    }
+
+    /**
+     * @param ArrayNodeDefinition $rootNode
+     */
+    private function addRoutesSection(ArrayNodeDefinition $rootNode)
+    {
+        $rootNode
+            ->children()
+            ->arrayNode('routes')
+                ->children()
+                    ->scalarNode('consume_assertion')
+                        ->isRequired()
+                        ->validate()
+                            ->ifTrue(function ($v) {
+                                return !is_string($v) || strlen($v) === 0;
+                            })
+                            ->thenInvalid('Consume assertion route must be a non-empty string')
+                        ->end()
+                    ->end()
+                    ->scalarNode('metadata')
+                        ->isRequired()
+                        ->validate()
+                            ->ifTrue(function ($v) {
+                                return !is_string($v) || strlen($v) === 0;
+                            })
+                            ->thenInvalid('Metadata route must be a non-empty string')
+                        ->end()
+                    ->end()
+                ->end()
+            ->end();
+    }
+
+    /**
+     * @param ArrayNodeDefinition $rootNode
+     */
+    private function addProvidersSection(ArrayNodeDefinition $rootNode)
+    {
+        /** @var ArrayNodeDefinition $protoType */
+        $protoType = $rootNode
+            ->children()
+                ->arrayNode('providers')
+                ->isRequired()
+                ->requiresAtLeastOneElement()
+                ->useAttributeAsKey('type')
+                ->prototype('array');
+
+        $protoType
+            ->children()
+                ->arrayNode('hosted')
+                    ->children()
+                        ->arrayNode('service_provider')
+                            ->children()
+                                ->scalarNode('public_key')
+                                    ->isRequired()
+                                    ->info('The absolute path to the public key used to sign AuthnRequests')
+                                ->end()
+                                ->scalarNode('private_key')
+                                    ->isRequired()
+                                    ->info('The absolute path to the private key used to sign AuthnRequests')
+                                ->end()
+                            ->end()
+                        ->end()
+                        ->arrayNode('identity_provider')
```

Comment:
This does not seem to correspond with https://github.com/SURFnet/Stepup-SelfService/pull/46/files#diff-7e86d2953bf50622c45b4d64a757b8ecR14 ?